lib/ansible/modules/cloud/google/gcpubsub.py

```diff
@@ -0,0 +1,314 @@
+#!/usr/bin/python
+# Copyright 2016 Google Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: gcpubsub
+version_added: "2.3"
+short_description: Create and Delete Topics/Subscriptions, Publish and pull messages on PubSub.
+description:
+    - Create and Delete Topics/Subscriptions, Publish and pull messages on PubSub.
+      See U(https://cloud.google.com/pubsub/docs) for an overview.
+requirements:
+  - "python >= 2.6"
+  - "google-auth >= 0.5.0"
+  - "google-cloud-pubsub >= 0.22.0"
+notes:
+  - Subscription pull happens before publish.  You cannot publish and pull in the same task.
+author:
+  - "Tom Melendez (@supertom) <tom@supertom.com>"
+options:
```

Comment:
The `state` option isn't listed here, and since it's got some special cases [in this comment](https://github.com/ansible/ansible/pull/19091/files#diff-527e1f4559a1720705ab1c1a7c6d294fR259) I think it's worth adding some extra explanation.